src/crimson/os/seastore/onode_manager/staged-fltree/node_extent_accessor.h

```diff
@@ -551,21 +548,12 @@ class NodeExtentAccessorT {
       return c.nm.retire_extent(c.t, to_discard
       ).handle_error_interruptible(
         eagain_iertr::pass_further{},
-        crimson::ct_error::input_output_error::assert_failure(
-            [FNAME, c, l_to_discard = to_discard->get_laddr(),
-             l_fresh = fresh_extent->get_laddr()] {
-          SUBERRORT(seastore_onode,
-              "EIO during retire -- to_disgard={}, fresh={}",
-              c.t, l_to_discard, l_fresh);
-        }),
-        crimson::ct_error::enoent::assert_failure(
-            [FNAME, c, l_to_discard = to_discard->get_laddr(),
-             l_fresh = fresh_extent->get_laddr()] {
-          SUBERRORT(seastore_onode,
-              "ENOENT during retire -- to_disgard={}, fresh={}",
-              c.t, l_to_discard, l_fresh);
-        })
-      );
+        crimson::ct_error::input_output_error::assert_failure(fmt::format(
+          "{} EIO during retire -- to_disgard={}, fresh={}",
+          FNAME, to_discard->get_laddr(), fresh_extent->get_laddr()).c_str()),
+        crimson::ct_error::enoent::assert_failure(fmt::format(
+          "{} EIO during retire -- to_disgard={}, fresh={}",
```

Comment:
nit, EIO -> ENOENT.

If the error handler can print the error type, the more general `assert_all()` can be used.

Reply:
> If the error handler

It can now :)
Missed the assert_all usecase - updated!